internal/thelma/render/resolver/local_resolver.go

```diff
@@ -96,3 +103,53 @@ func (r *localResolverImpl) getChart(chartName string) (source.Chart, error) {
 func (r *localResolverImpl) chartSourcePath(chartName string) string {
 	return path.Join(r.sourceDir, chartName)
 }
+
+// determineDependenciesToUpdate is used to add support for lack of handling for multi-layer depedencies
+// in helm upgrade. It performs a BFS traversal of the dependency graph for a given chart and outputs
+// a topologically sorted list of charts to run helm depedency update upon
+func (r *localResolverImpl) determineDepedenciesToUpdate(chart source.Chart) ([]source.Chart, error) {
```

Comment:
nit: typo `Depedencies`

Reply:
I spelled that wrong so many times while doing this lol